crucible-llvm/src/Lang/Crucible/LLVM/MemModel.hs

```diff
@@ -469,6 +478,35 @@ evalStmt sym = eval
     do mem <- getMem mvar
        liftIO $ doPtrSubtract sym mem x y
 
+-- | Utility function for testing if @xs@ is a context with
+--   `prefix` as a prefix, and computing the tail of xs
+--   not in the prefix, if so.
+splitPrefix :: forall xs prefix a.
```

Comment:
Could/should this go into parameterized-utils?

Reply:
It probably could, if we generalized it slightly over arbitrary `Assignment`s rather than just `CtxRepr`s.

Should it? Yeah, maybe... I guess it's probably useful enough to upstream it.